lib/job_queue/ext_job.cpp

```diff
@@ -866,17 +866,24 @@ static void __fprintf_python_arg_types(FILE * stream,
 }
 
 
-void ext_job_json_fprintf(const ext_job_type * ext_job, FILE * stream, const subst_list_type * global_args) {
+void ext_job_json_fprintf(const ext_job_type * ext_job, int job_index, FILE * stream, const subst_list_type * global_args) {
   const char * null_value = "null";
+  
+  char * file_stdout_index = NULL;
+  char * file_stderr_index = NULL;
+  
+  file_stdout_index = util_alloc_sprintf("%s.%d",ext_job->stdout_file, job_index);
+  file_stderr_index = util_alloc_sprintf("%s.%d",ext_job->stderr_file, job_index);
```

Comment:
This is leaking memory, you also need to add 
```
free( file_stdout_index );
free( file_stderr_index );
```
after you are done using the string.